third_party/proton/README.md

```diff
@@ -52,7 +52,32 @@ proton.activate(session_id)
 # Write out the profile data and finalize the profiler
 proton.finalize()
 ```
+### Advanced features
+In addition to profiling, Proton also incorporates MLIR/LLVM based compiler based instrumentation passes to get Triton level analysis
+and optimization information. This feature is under active development and the list of available passes is expected to grow.
 
+#### Available passes
+print-mem-spaces: this pass prints the load and store address spaces (e.g. global, flat, shared) chosen by the compiler and attributes back to Triton source information.
+
+Example usage with the Proton matmul tutorial:
+```bash
+$ proton --instrument=print-mem-spaces matmul.py
+0     matmul_kernel     matmul.py:180:20     SHARED     STORE
+1     matmul_kernel     matmul.py:181:20     SHARED     STORE
+2     matmul_kernel     matmul.py:180:20     SHARED     LOAD
+3     matmul_kernel     matmul.py:181:20     SHARED     LOAD
+
+matmul-performance:
```

Comment:
We can remove the performance numbers from here and below.

Reply:
Removed. Good point.